publisher-sdk/src/main/java/com/criteo/publisher/adview/MraidInteractor.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.adview
+
+import android.webkit.WebView
+import com.criteo.publisher.annotation.OpenForTesting
+
+@OpenForTesting
+internal class MraidInteractor(private val webView: WebView) {
+
+  fun notifyReady() {
+    callOnMraidObject("notifyReady()")
+  }
+
+  fun notifyError(message: String, action: String? = null) {
+    val actionString = if (action == null) "" else ", \"$action\""
+    callOnMraidObject("notifyError(\"$message\"$actionString)")
+  }
+
+  private fun callOnMraidObject(jsCode: String) {
+    webView.evaluateJavascript("window.mraid.$jsCode", null)
```

Comment:
Another option is to log debug all calls so that you could clearly see if there is any error when interacting with MRAID.

Reply:
Added call log to simplify debugging in case of problem